tiledb/sm/filesystem/s3.h

```diff
@@ -505,6 +529,47 @@ class S3 {
     int upload_part_num;
   };
 
+  /**
+   * This type holds the location and the size of an intermediate chunk on S3
+   * These chunks are used with remote global writes according to the
+   * algorithm below:
+   *
+   * Intermediate chunks are persisted across cloud executors via serialization
+   * When a new write arrives via global_order_write(URI):
+   * - look if there is a multipart state for the URI, if not,
+   *   create one and initiate multipart request
+   * - look at the sizes of all intermediate chunks persisted before
+   *   for this URI
+   * - If sum(intermediate_chunks[i].size) + current_chunk_size < 5MB
+   *   - persist the new chunk as an intermediate chunk on s3 under
+   *     fragment_uri/__gow_chunks/buffer_name_intID
+   * - Else
+   *   a) read all previous intermediate chunks,
+   *   b) merge them in memory including the current chunk,
+   *   c) upload the merged buffer as a new part using multipart upload
+   *   d) delete all intermediate chunks under __gow_chunks/ for this URI
+   *   e) clear the intermediate chunks from multipart_upload_states[uri]
+   * When the global order write is done and the buffer file is finalized
+   * via finalize_and_flush_object(uri):
+   *   a) read all intermediate chunks (not uploaded as >5mbs multipart parts)
```

Comment:
Should we go back to - instead of letters here? Letters make me think this is still part of the If/Else above.